### PR TITLE
Alternative maybe is text

### PR DESCRIPTION
**Pull request by loesinghaus**
Changed maybe_is_text() to calculate the entropy without spaces to avoid discarding of valid papers due to title pages or tables. Also use the first five text chunks instead of just the first text chunk in case of other potential issues peculiar to title pages.

Closes https://github.com/Future-House/paper-qa/issues/710

### CODE REVIEW

File: paperqa/utils.py [hunk 1]
```diff
@@ -52,10 +52,12 @@ def name_in_text(name: str, text: str) -> bool:
 def maybe_is_text(s: str, thresh: float = 2.5) -> bool:
     if not s:
         return False
+
     # Calculate the entropy of the string
     entropy = 0.0
+    s_wo_spaces = s.replace(" ", "")
```
Comment (jamesbraza) [inline paperqa/utils.py:58]: Can you update the comment two lines up to mention the heuristic for calculating entropy?

File: paperqa/docs.py [hunk 2]
```diff
@@ -402,7 +402,7 @@ async def aadd(  # noqa: PLR0912
             or len(texts[0].text) < 10  # noqa: PLR2004
             or (
                 not parse_config.disable_doc_valid_check
-                and not maybe_is_text(texts[0].text)
+                and not maybe_is_text("".join(text.text for text in texts[:5]))
```
Comment (jamesbraza) [inline paperqa/docs.py:405]: Can you add a comment here saying why `:5`?

**Review by jamesbraza — COMMENTED** [review]
Looking good, thank you!

File: paperqa/docs.py (hunk 2 above)
Comment (loesinghaus) [reply paperqa/docs.py:406]: Done.

File: paperqa/utils.py (hunk 1 above)
Comment (loesinghaus) [reply paperqa/utils.py:63]: I added a docstring with the reasoning for entropy calculation and leaving out spaces.

**Review by jamesbraza — APPROVED** [review]
LGTM

**Comment by loesinghaus** [comment]
Alright :) Running the tests on my setup was also sort of difficult because they tend to trip my rate limits (need to spend more money on OpenAI, I guess).

**Comment by jamesbraza** [comment]
I am not sure why [the CI run](https://github.com/Future-House/paper-qa/actions/runs/12127798131/job/33813739247?pr=717) is seeing:

```none
httpx.LocalProtocolError: Illegal header value b'Bearer '
```

Ugh our contributor developer experience is trash.

Anyways, tests pass locally on my Mac, so I am going to merge

**Comment by loesinghaus** [comment]
No trouble, thanks for merging this one.

**Pull request merged**